### ISSUE TITLE
Cannot replicate FID score for SD unlearning.

### ISSUE THREAD

**Issue by anseryuer** [OPTML-Group/Unlearn-Saliency#23]
Hi!

Recently I am trying to replicate your score. While I am working on the part of SD unlearning, I generate mask, then trained model with Salun, after that I generate a certain number of images to do further evaluation of the FID metrics.

Here is your code for computing FID

```# https://huggingface.co/docs/diffusers/conceptual/evaluation

import argparse

import torch
from dataset import setup_fid_data
from torchmetrics.image.fid import FID


def compute_fid(class_to_forget, path, image_size):
    fid = FID(feature=64)
    real_set, fake_set = setup_fid_data(class_to_forget, path, image_size)
    real_images = torch.stack(real_set).to(torch.uint8).cpu()
    fake_images = torch.stack(fake_set).to(torch.uint8).cpu()

    fid.update(real_images, real=True)  # doctest: +SKIP
    fid.update(fake_images, real=False)  # doctest: +SKIP
    print(fid.compute())  # doctest: +SKIP
```

Here is your code in dataset.py for setting up the dataset:

```def setup_fid_data(class_to_forget, path, image_size, interpolation="bicubic"):
    interpolation = INTERPOLATIONS[interpolation]
    transform = get_transform(interpolation, image_size)

    real_set = Fake_Imagenette(
        "imagenette_without_label_6", class_to_forget, transform=transform
    )
    real_set = [data[0] for data in real_set]

    fake_set = Fake_Imagenette(path, class_to_forget, transform=transform)
    fake_set = [data[0] for data in fake_set]

    return real_set, fake_set
```

I have 2 questions here.

First, you do need some real images to compute. While running the code, you need to have this folder imagenette_without_label_6 with all the real images in it. However, how do you decide which real image to use, do you use every image within the imagenette? Is there any code to generate the real image dataset?

Second, what is this 

```
    real_set = Fake_Imagenette(
        "imagenette_without_label_6", class_to_forget, transform=transform
    )
```

Why there is imagenette_without_label_6 here in the code?

**Comment by a-F1** [comment]
Thank you for your appreciation and recognition of our work.

Regarding Q1 about real images, we randomly selected 50 images from each class in the Imagenette dataset, totaling 500 images as the real images.

As for Q2, `imagenette_without_label_6` is the file path where the real images are stored. You can modify this path to the appropriate one for your setup.

If you have any other questions, please feel free to contact us.

**Comment by a-F1** [comment]
Thank you for your appreciation and recognition of our work.

Regarding Q1 about real images, we randomly selected 50 images from each class in the Imagenette dataset, totaling 500 images as the real images.

As for Q2, imagenette_without_label_6 is the file path where the real images are stored. You can modify this path to the appropriate one for your setup.

If you have any other questions, please feel free to contact us.

**Comment by a-F1** [comment]
The issue has been closed. If you have any further questions, please feel free to contact us.

**Comment by k-zhai** [comment]
You're using 64 features for FID, why not use 2048 like everyone else?

**Comment by anseryuer** [comment]
> You're using 64 features for FID, why not use 2048 like everyone else?

64 is the number in salun code.

And if you do run the code, there is only 900 generated images and 450 real images to compare. So 64 is enough.
